config/config.go

```diff
@@ -204,7 +206,25 @@ func (c *Config) Authenticate(r *http.Request) error {
 	if err != nil {
 		return err
 	}
-	return c.auth(r)
+	return c.credentialsProvider.SetHeaders(r)
+}
+
+// Authenticate returns an OAuth token for the current configuration.
+// It will return an error if the CredentialsStrategy does not support OAuth tokens.
+func (c *Config) GetToken() (*oauth2.Token, error) {
```

Comment:
This is the method that will be used in the follow-up PR?

Reply:
This is already being used in `GetOAuthToken` method in the Workspace Client
https://github.com/databricks/databricks-sdk-go/pull/886/files#diff-75f383f34048f57e9041c3a70e938f38345a52ca6473a7790c7069d801b9bacfR980
We will use that one in the follow-up PR